legend-engine-xt-openapi-generation/src/main/java/org/finos/legend/engine/generation/OpenApiArtifactGenerationExtension.java

```diff
@@ -0,0 +1,56 @@
+//  Copyright 2022 Goldman Sachs
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package org.finos.legend.engine.generation;
+
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.dsl.generation.extension.Artifact;
+import org.finos.legend.engine.language.pure.dsl.generation.extension.ArtifactGenerationExtension;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContextData;
+import org.finos.legend.pure.generated.Root_meta_external_format_openapi_metamodel_Server_Impl;
+import org.finos.legend.pure.generated.Root_meta_legend_service_metamodel_Service;
+import org.finos.legend.pure.generated.core_external_format_openapi_generation;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.PackageableElement;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.finos.legend.engine.shared.core.operational.Assert.assertTrue;
+
+public class OpenApiArtifactGenerationExtension implements ArtifactGenerationExtension
+{
+
+    private static final String ROOT_PATH = "openapi_spec";
+
+    @Override
+    public String getKey()
+    {
+        return ROOT_PATH;
+    }
+
+    @Override
+    public boolean canGenerate(PackageableElement element)
+    {
+        return element instanceof Root_meta_legend_service_metamodel_Service;
+    }
+
+    @Override
+    public List<Artifact> generate(PackageableElement element, PureModel pureModel, PureModelContextData data, String clientVersion)
+    {
+        assertTrue(this.canGenerate(element), () -> "OpenApi extension only supports service elements");
+        String result = core_external_format_openapi_generation.Root_meta_external_format_openapi_generation_serviceToOpenApi_Service_1__Server_1__String_1_((Root_meta_legend_service_metamodel_Service) element, new Root_meta_external_format_openapi_metamodel_Server_Impl("")._url("localhost"), pureModel.getExecutionSupport());
+        Artifact output = new Artifact(result, element.getName() + "_openapi.json", "json");
```

Comment:
One thing worth noting is that the generator is already qualifying the generated file by the path and the "key"

For example, `abc::abc::ABCService` will get this file under:  `abc/abc/ABCService/openapi_spec/{your_filename_here}`

Hence, you can make this name more vanilla to make thing easier for the system that will like to use these.  For example, you can just set the name to `spec.json`.

Reply:
Makes sense, will update the file name suffix to something simpler like spec.json

Reply:
Also, can we get the url/host of legend service somehow? Hardcoding localhost here doesn't seem right. Can we read any configuration values for host?

Reply:
Not really.  These files are generated during the Studio project pipelines on GitLab.  There is no server.  Can we use some sort of template, like ${HOST} or something that reflects that is a value that needs to change.  How to change it and who will need to change it is a different discussion.

Reply:
Thanks for the suggestion. Applied the changes along with logs for failure